utils/s3/client.cc

```diff
@@ -307,17 +307,15 @@ client::group_client& client::find_or_create_client() {
 
 future<> client::make_request(http::request req, http::experimental::client::reply_handler handle, std::optional<http::reply::status_type> expected, seastar::abort_source* as) {
     co_await authorize(req);
-    auto& gc = find_or_create_client();
-    co_await gc.retryable_client.make_request(std::move(req), std::move(handle), expected, as);
+    co_await find_or_create_client().retryable_client.make_request(std::move(req), std::move(handle), expected, as);
```

Comment:
Is there a functional change here?
If not, don't mix the code cleanups with functional patches.
It just adds noise making review harder